docs/concepts/overview/working-with-objects/common-labels-annotations.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Recommended Labels and Annotations
+---
+
+{% capture overview %}
+Many tools can be used to visualize and manage Kubernetes objects. The tools used
+to manage objects goes beyond kubectl and the dashboard. To enable interoperability
+between tools, a common set of labels and annotation allows objects to be described
+in a common manner that all tools can understand.
+
+In addition to supporting tooling, the recommended labels and annotations describe
+applications, in a generic sense, in a way that can queried or are otherwise useful
+when operating applications.
+{% endcapture %}
+
+{% capture body %}
+The recommended object metadata is broken apart so that some of it is labels while
+other parts are annotations. Labels are shorter and can be used when querying
+the Kubernetes API. Annotations allow for longer data that's not meant to be
+queried.
+
+The metadata is organized around the concept of an Application. Kubernetes is not
+a Platform as a Service and doesn't have or enforce a formal notion of an application.
+Instead, applications are informal, described with metadata, and the definition of what an
+application contains is loose.
+
+Note, these are recommended labels and annotations. They aid the use of managing
+applications but are not required for any core tooling to work.
+
+One common theme you'll notice about the labels and annotations is that they
+contain a prefix of `app.kubernetes.io`. Labels and annotations without a prefix
+are considered private to users. These are meant to be common labels that do not
+interfere with existing or custom user labels. To accomplish this a prefix is
+used.
+
+## Labels
+
+| Key                                 | Description           | Example  | Type |
+| ----------------------------------- | --------------------- | -------- | ---- |
+| `app.kubernetes.io/name`              | The name of the application stack | `wordpress` | string |
+| `app.kubernetes.io/version`           | The current version of the running application (e.g., a semantic version, revision hash, etc.) | `v1.2.3` | string |
```

Comment:
It's unclear whether this version only referer to the upstream component, or maybe it should include packaging version.

Reply:
I also think the version labels are going to be problematic. Guidance should be that controllers and services should not select on the versions. 

To avoid confusion, they might be more appropriate to represent as annotations.